app/models/certificate.rb

```diff
@@ -14,17 +14,28 @@ class Certificate < Sequel::Model
   one_to_many :measure_conditions, key: %i[certificate_type_code certificate_code],
                                    primary_key: %i[certificate_type_code certificate_code]
 
+  one_to_many :certificate_type_descriptions, key: %i[certificate_type_code],
+                                              primary_key: %i[certificate_type_code]
+
   many_to_many :measures, join_table: :measure_conditions,
                           left_key: %i[certificate_code certificate_type_code],
                           right_key: :measure_sid
 
+  one_to_many :certificate_types, key: :certificate_type_code,
+                                  primary_key: :certificate_type_code do |ds|
+    ds.with_actual(CertificateType)
+  end
+
   def certificate_description
     certificate_descriptions.first
   end
 
-  one_to_many :certificate_types, key: :certificate_type_code,
-                                  primary_key: :certificate_type_code do |ds|
-    ds.with_actual(CertificateType)
+  def certificate_type_description
+    certificate_type_descriptions.first
+  end
+
+  def certificate_description_period
+    CertificateDescriptionPeriod[certificate_description.certificate_description_period_sid]
```

Comment:
This feels and reads wrong :eyes:

Reply:
Removed, and added the new relationship to `certificate_description_period`